### PR TITLE
[shelly] Add check and ThingStatus for local IP issue (APIPA)

### PR DESCRIPTION
**Pull request by markus7017**
The binding checks now the local ip to be != 169.254.x.x, which is the result of a failed DHCP assignment. The framework reports this in the list of interfaces and the binding had send this to the device as part of the callback url, which resulted in device functioning in general, but the binding is not receiving any events. Now the user sees a WARNING is the log, which is a clear indicator of a setup to be fixed.

### CODE REVIEW

**Review by lsiepel — CHANGES_REQUESTED** [review]
Minor comments.

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java [hunk 1]
```diff
@@ -1008,6 +1008,13 @@ protected void initializeThingConfig() {
         config.serviceName = getString(properties.get(PROPERTY_SERVICE_NAME));
         config.localIp = bindingConfig.localIP;
         config.localPort = String.valueOf(bindingConfig.httpPort);
+        if (config.localIp.startsWith("169.254")) {
+            logger.warn("{}: An invalid local ip address {} was detected, check system and openHAB network settings",
```
Comment (lsiepel) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:1012]: As the thing state is allready set, there is no need to log a warn message.

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java [hunk 2]
```diff
@@ -1008,6 +1008,13 @@ protected void initializeThingConfig() {
...
+                    thingName, config.localIp);
+            setThingOffline(ThingStatusDetail.CONFIGURATION_ERROR, "config-status.error.network-config");
```
[the remark this answers is not in the document]
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java]: changed

**Comment by markus7017** [comment]
@lsiepel requested changes applied

File: bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly.properties [hunk 3]
```diff
@@ -11,6 +11,7 @@ addon.shelly.config.autoCoIoT.label = Auto-CoIoT
 addon.shelly.config.autoCoIoT.description = If enabled CoIoT will be automatically used when the devices runs a firmware version 1.6 or newer; false: Use thing configuration to enabled/disable CoIoT events.  
 
 # Config status messages
+message.config-status.error.network-config = Invalid system / openHAB network configuration was detected.
```
[the remark this answers is not in the document]
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/resources/OH-INF/i18n/shelly.properties]: changed

**Review by lsiepel — APPROVED** [review]
Thanks, LGTM

**Pull request merged**